zk-token-sdk/src/instruction/transfer/with_fee.rs

```diff
@@ -871,47 +871,5 @@ mod test {
         .unwrap();
 
         assert!(fee_data.verify_proof().is_err());
-
-        // auditor pubkey invalid
-        let destination_keypair = ElGamalKeypair::new_rand();
-        let destination_pubkey = destination_keypair.pubkey();
-
-        let auditor_pubkey = pod::ElGamalPubkey::zeroed().try_into().unwrap();
-
-        let withdraw_withheld_authority_keypair = ElGamalKeypair::new_rand();
-        let withdraw_withheld_authority_pubkey = withdraw_withheld_authority_keypair.pubkey();
-
-        let fee_data = TransferWithFeeData::new(
-            transfer_amount,
-            (spendable_balance, &spendable_ciphertext),
-            &source_keypair,
-            (destination_pubkey, &auditor_pubkey),
-            fee_parameters,
-            withdraw_withheld_authority_pubkey,
-        )
-        .unwrap();
-
-        assert!(fee_data.verify_proof().is_err());
-
-        // withdraw authority invalid
-        let destination_keypair = ElGamalKeypair::new_rand();
-        let destination_pubkey = destination_keypair.pubkey();
-
-        let auditor_keypair = ElGamalKeypair::new_rand();
-        let auditor_pubkey = auditor_keypair.pubkey();
-
-        let withdraw_withheld_authority_pubkey = pod::ElGamalPubkey::zeroed().try_into().unwrap();
-
-        let fee_data = TransferWithFeeData::new(
-            transfer_amount,
-            (spendable_balance, &spendable_ciphertext),
-            &source_keypair,
-            (destination_pubkey, auditor_pubkey),
-            fee_parameters,
-            &withdraw_withheld_authority_pubkey,
-        )
-        .unwrap();
-
-        assert!(fee_data.verify_proof().is_err());
```

Comment:
Did you mean to remove this too? I thought the change was only for an invalid auditor, not an invalid withdraw withheld authority. Unless the withdraw withheld authority behaves like the auditor at the base layer

Reply:
Yeah, so this is very bad design in terms of naming...

The ciphertext validity proof is used to certify that a "grouped" ciphertext is well-formed. For transfers (without fee), this is only used for ciphertext that holds the transfer amount. The transfer amount is encrypted under the destination and auditor ElGamal public key.

For transfer with fees, this proof is used on both the transfer amount ciphertext and the fee ciphertext. For the case of fee ciphertext, the fee amount is encrypted under the destination and withdraw withheld authority public key.

This PR is loosening the restriction so that the second associated public key with the proof can be an all-zero public key. For the case of verifying the transfer amount ciphertext, the second public key is the auditor, but for the case of fee ciphertext, the second public key is the withdraw withheld authority public key. So this test necessarily has to be removed. The need to loosen the restriction for withdraw withheld authority is not needed (since withdraw withheld authority public key should always be randomly generated and never be all zero), but since this restriction was an extra safety measure anyway, this should be fine.

Reply:
Ok great, just wanted to make sure, thanks for the explanation!